### DISCUSSION TITLE
Pass record from ViewRecord to embedded Widget

### DISCUSSION THREAD

**Discussion by PierreSchwang** [filamentphp/filament#5195] [Help] [answered]
My goal is to embed a TableWidget containing related data of the widget from the ViewRecord-page where the widget is embedded.
Is there any way to achieve that?

**Comment by danharrin** [comment, marked as answer]
https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record

**Reply by PierreSchwang** [reply]
Tried that, but the record was still null. Gonna try that again this evening and provide a minimal example to reproduce the issue

**Reply by danharrin** [reply]
Do you have a custom view for this page?

**Reply by PierreSchwang** [reply]
Ah, I guess my issue was the dd directly in the constructor - That was obviously way too early. Works perfect, thank you!

**Reply by MasterOfWeb** [reply]
I have probably the same problem. Can you provide a code example, how you got the record I in the getTableQuery? 

`    public ?Model $record = null;
    
    public function getTableQuery(): Builder 
    {   
        dd($this->record); // -> null
    …
`
